flowey/flowey_lib_common/src/git_merge_commit.rs

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+//! Gets the merge commit of a PR to base branch
+
+use flowey::node::prelude::*;
+
+flowey_request! {
+    pub struct Request {
+        pub repo_path: ReadVar<PathBuf>,
+        pub merge_commit: WriteVar<String>,
+        pub base_branch: String,
+    }
+}
+
+new_simple_flow_node!(struct Node);
+
+impl SimpleFlowNode for Node {
+    type Request = Request;
+
+    fn imports(_ctx: &mut ImportCtx<'_>) {}
```

Comment:
note that this node assumes the existince of `git` on the machine. this might not be a problem in practice, but for max correctness, you should take a dep on `install_git`

Reply:
this ties into https://github.com/microsoft/openvmm/pull/475#discussion_r1937937612, insofar as you should be modeling that constraint at the pipeline level, vs. trying to infer the pipeline scenario within the job, based on the downstream parameters that are being passed in.